java-squid/src/main/java/org/sonar/java/ast/parser/JavaNodeBuilder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.ast.parser;
+
+import com.google.common.collect.Lists;
+import com.sonar.sslr.api.GenericTokenType;
+import com.sonar.sslr.api.Rule;
+import com.sonar.sslr.api.Token;
+import com.sonar.sslr.api.TokenType;
+import com.sonar.sslr.api.Trivia;
+import com.sonar.sslr.api.typed.Input;
+import com.sonar.sslr.api.typed.NodeBuilder;
+import org.sonar.java.model.InternalSyntaxSpacing;
+import org.sonar.java.model.InternalSyntaxToken;
+import org.sonar.java.model.InternalSyntaxTrivia;
+import org.sonar.plugins.java.api.tree.SyntaxTrivia;
+import org.sonar.sslr.grammar.GrammarRuleKey;
+
+import java.util.List;
+
+public class JavaNodeBuilder implements NodeBuilder {
+
+  @Override
+  public Object createNonTerminal(GrammarRuleKey ruleKey, Rule rule, List<Object> children, int startIndex, int endIndex) {
+    for (Object child : children) {
+      if (child instanceof InternalSyntaxToken) {
+        InternalSyntaxToken syntaxToken = (InternalSyntaxToken) child;
+        syntaxToken.setGrammarRuleKey(ruleKey);
+        return child;
+      }
+    }
+    return new InternalSyntaxSpacing(startIndex, endIndex);
+  }
+
+  @Override
+  public Object createTerminal(Input input, int startIndex, int endIndex, List<Trivia> trivias, TokenType type) {
+    boolean isEof = GenericTokenType.EOF == type;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)